core/src/main/java/io/substrait/expression/Expression.java

```diff
@@ -662,6 +669,30 @@ public <R, C extends VisitationContext, E extends Throwable> R accept(
     }
   }
 
+  @Value.Immutable
+  abstract class StructNested implements Nested {
+    public abstract List<Expression> fields();
+
+    @Override
+    public Type getType() {
+      return Type.withNullability(false)
```

Comment:
```suggestion
      return Type.withNullability(nullable())
```
We should respect the nullability instead of dropping that info.